core/src/main/kotlin/entity/component/SelectMenuComponent.kt

```diff
@@ -12,10 +13,7 @@ import dev.kord.core.entity.interaction.SelectMenuInteraction
  * An interactive dropdown menu rendered on a [Message] that consists of multiple [options].
  */
 
-public class SelectMenuComponent(override val data: ComponentData) : Component {
-
-    override val type: ComponentType.SelectMenu
-        get() = ComponentType.SelectMenu
+public open class SelectMenuComponent(override val data: ComponentData) : Component {
```

Comment:
```suggestion
public open class SelectMenuComponent
@Deprecated(
    "This will be made a sealed class in the future, please stop using this constructor. You can instead use the " +
            "constructor of one of the subtypes.",
    level = WARNING,
)
public constructor (override val data: ComponentData) : Component {
```
We will increase the deprecation level and then replace it with a sealed class.